### PR TITLE
update config args to map[string]interface{} to extend the configs

### PR DESCRIPTION
**Pull request by Thor-wl**
This update will help extend the configurations for the furture. Besides, it will be used in the following PR for the rescheduling feature. The configuration example can be referred to https://github.com/volcano-sh/volcano/blob/3a7007f0e4bfb5b82c349409ecafc49bd09bda9e/docs/design/rescheduling.md
Signed-off-by: Thor-wl <13164644535@163.com>

### CODE REVIEW

File: pkg/scheduler/framework/arguments.go [hunk 1]
```diff
@@ -34,13 +33,14 @@ func (a Arguments) GetInt(ptr *int, key string) {
 	}
 
 	argv, ok := a[key]
-	if !ok || argv == "" {
+	if !ok {
 		return
 	}
 
-	value, err := strconv.Atoi(argv)
+	valueStr, ok := argv.(string)
```
Comment (hwdef) [inline pkg/scheduler/framework/arguments.go:40]: Is this ok used?

File: pkg/scheduler/framework/arguments.go [hunk 2]
```diff
@@ -54,13 +54,14 @@ func (a Arguments) GetFloat64(ptr *float64, key string) {
 	}
 
 	argv, ok := a[key]
-	if !ok || len(argv) == 0 {
+	if !ok {
 		return
 	}
 
-	value, err := strconv.ParseFloat(argv, 64)
+	valueStr, ok := argv.(string)
```
Comment (hwdef) [inline pkg/scheduler/framework/arguments.go:61]: here too

File: pkg/scheduler/framework/arguments.go [hunk 3]
```diff
@@ -74,13 +75,14 @@ func (a Arguments) GetBool(ptr *bool, key string) {
 	}
 
 	argv, ok := a[key]
-	if !ok || argv == "" {
+	if !ok {
 		return
 	}
 
-	value, err := strconv.ParseBool(argv)
+	valueStr, ok := argv.(string)
```
Comment (hwdef) [inline pkg/scheduler/framework/arguments.go:82]: here too

File: pkg/scheduler/framework/arguments.go (hunk 1 above)
Comment (Thor-wl) [reply pkg/scheduler/framework/arguments.go]: No, let me remove unused variables.

File: pkg/scheduler/conf/scheduler_conf.go [hunk 4]
```diff
@@ -36,7 +36,7 @@ type Configuration struct {
 	// Name is name of action
 	Name string `yaml:"name"`
 	// Arguments defines the different arguments that can be given to specified action
-	Arguments map[string]string `yaml:"arguments"`
+	Arguments map[string]interface{} `yaml:"arguments"`
```
Comment (william-wang) [inline pkg/scheduler/conf/scheduler_conf.go:39]: We have to consider how to keep compatible with user developed plugins.

Comment (Thor-wl) [reply pkg/scheduler/conf/scheduler_conf.go:39]: > We have to consider how to keep compatible with user developed plugins.

This update will not influence users who make use of the built-in plugins. A little work to match the interface is needed for users who implement custom plugins with arguments. I've no idea how to support that gracefully. Hope to see some advices.

Comment (hwdef) [reply pkg/scheduler/conf/scheduler_conf.go:39]: Wait until go1.18, we can try to use go's generics

Comment (Thor-wl) [reply pkg/scheduler/conf/scheduler_conf.go:39]: > Wait until go1.18, we can try to use go's generics

@william-wang means the update from `string` to `interface` will introduces some code modifications in custom plugins, for example, the value should be converted to `string` with assert after getting the value from `Arguments`. I agree with that finding a way with no code modification for users is the best. If we cannot achieve that, the update is also acceptable for two factors: 
1. Users who make use of custom plugins are few as to the feedback from issues. 
2. The modification is of little work.

**Review by william-wang — APPROVED** [review]
/lgtm

**Pull request merged**